aiohttp/client_reqrep.py

```diff
@@ -931,8 +931,10 @@ def release(self) -> Any:
         return noop()
 
     def raise_for_status(self) -> None:
+        # self.reason is only None if self.start() hasn't been called:
+        if self.reason is None:
+            raise TypeError("Expected a Reason-Phrase as a string")
```

Comment:
An exception type and especially text is meaningless for an end user.
The text describes a consequence (the `reason` field is not set) but hides an actual problem.

Moreover, assertion exists only for sake of `mypy`.
If you don't like the assertion -- please find another way to satisfy mypy.

`assert self.reason is not None` could be the simplest possible fix, even no test is needed for the case.

Reply:
Well, a test would be nice but please use a functional test for an empty `reason`.
https://github.com/aio-libs/aiohttp/blob/master/tests/test_client_functional.py can give an inspiration how a functional client test looks like

Reply:
I’ll switch it to `assert self.reason is not None` if that’s the desired outcome.